inc/ThirdParty/Hostings/HostResolver.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace WP_Rocket\ThirdParty\Hostings;
+
+/**
+ * Host Resolver.
+ *
+ * @since 3.6.3
+ */
+class HostResolver {
+
+	/**
+	 * Name of the current host service.
+	 *
+	 * @var string
+	 */
+	private static $hostname = '';
+
+	/**
+	 * Get the name of an identifiable hosting service.
+	 *
+	 * @since 3.6.3
+	 *
+	 * @param bool $ignore_cached_hostname (optional) Don't use cached hostname when true.
+	 *
+	 * @return string Name of the hosting service or '' if no service is recognized.
+	 */
+	public static function get_host_service( $ignore_cached_hostname = false ) {
+
+		if ( ! $ignore_cached_hostname && self::$hostname ) {
```

Comment:
The hostname is a string. Avoid boolean type juggling by checking it either via `empty()` or `'' !== self::$hostname` or even adding a `has` method.

Reply:
Done in https://github.com/wp-media/wp-rocket/pull/2874/commits/9a9a5970182e068a13e9703d1860ef151813e05a